compiler/back_end/cpp/header_generator.py

```diff
@@ -1244,24 +1272,26 @@ def _generate_enum_definition(type_ir):
   enum_type = _cpp_integer_type_for_enum(max_bits, is_signed)
   for value in type_ir.enumeration.value:
     numeric_value = ir_util.constant_value(value.value)
-    enum_values.append(
-        code_template.format_template(_TEMPLATES.enum_value,
-                                      name=value.name.name.text,
-                                      value=_render_integer(numeric_value)))
-    enum_from_string_statements.append(
-        code_template.format_template(_TEMPLATES.enum_from_name_case,
-                                      enum=type_ir.name.name.text,
-                                      name=value.name.name.text))
-    if numeric_value not in previously_seen_numeric_values:
-      string_from_enum_statements.append(
-          code_template.format_template(_TEMPLATES.name_from_enum_case,
+    enum_value_names = _get_enum_value_names(value)
+    for enum_value_name in enum_value_names:
+      enum_values.append(
+          code_template.format_template(_TEMPLATES.enum_value,
+                                        name=enum_value_name,
+                                        value=_render_integer(numeric_value)))
+      enum_from_string_statements.append(
```

Comment:
I'm not sure whether `TryToGetEnumFromName` and `TryToGetNameFromEnum` should be affected by `[(cpp) enum_case]`.  I am open to arguments either way, but I'm inclined to say that they should *not* be affected, and that there should be new functions that perform the same functions but use the C++ names.

Reply:
I think that's fair - the emboss name is the canonical name, after all, and the C++ name is just a back-end detail. I think there may still be value in having conversion function(s) for the C++-specific name, but that can be added later if there's a specific use case for it.